### PR TITLE
Functionality for NDK logs, warnings and errors to show in the Unity console

### PR DESCRIPTION
**Pull request by Div-Inworld**
Unity Console logging

### CODE REVIEW

**Comment by Matt-Carey** [comment]
I think we should try to get rid of UNITY/UNREAL specifics in NDK.
Started stripping this out here: https://github.com/inworld-ai/inworld-ndk/tree/remove-ue-specific-build
We can use callback instead of what I did above, but can we make this generic over UNITY specific so it can be used anywhere?

File: src/Utils/Log.cpp [hunk 1]
```diff
@@ -21,6 +21,14 @@ namespace Inworld
 		g_SessionId = "Unknown";
 	}
 
+#if INWORLD_UNITY
+
+	void LogSetUnityLogCallback(void(*callback)(const char*, int severity))
```
Comment (art-vozniuk) [inline src/Utils/Log.cpp:26]: a bit clumsy that one argument has name, and another doesn't. add 'message' or remove 'severinity' pls

File: src/Utils/Log.h [hunk 2]
```diff
@@ -42,6 +39,12 @@ namespace Inworld
 	INWORLDAINDK_API void LogSetSessionId(const std::string Id);
 	INWORLDAINDK_API void LogClearSessionId();
 
+#if INWORLD_UNITY
+	INWORLDAINDK_API void LogSetUnityLogCallback(void(*callback)(const char*, int severity));
+	inline std::function<void(const char *, int severity)> UnityLoggerCallback;
```
Comment (art-vozniuk) [inline src/Utils/Log.h:44]: here as well

**Review by art-vozniuk — APPROVED** [review]

File: src/Utils/Log.cpp (hunk 1 above)
Comment (Div-Inworld) [reply src/Utils/Log.cpp:26]: Will do! Code was generated by copilot 😅😅

**Pull request merged**